File: modelos ESPE/P3_D1_NRC9901_Reyes_David/c++/P3_D1_NRC9901_Reyes_David_No_1.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

using namespace std;

int main()
{
    int n, m;
    cout << "Ingrese el tamano de la matriz (nxm): ";
    cin >> n >> m;

    int matriz[n][m];
    srand(time(NULL));
    for(int i = 0; i < n; i++)
    {
        for(int j = 0; j < m; j++)
        {
            matriz[i][j] = rand() % 2;
        }
    }

    cout << "   ";
    for(int i = 0; i < m; i++)
    {
        cout << char('A' + i) << " ";
    }
    cout << endl;

    for(int i = 0; i < n; i++)
    {
        cout << char('A' + i) << "  ";
        for(int j = 0; j < m; j++)
        {
            cout << matriz[i][j] << " ";
        }
        cout << endl;
    }

    int lazos = 0;
    int aristas = 0;
    int grado[n] = {0};

    for(int i = 0; i < n; i++)
    {
        for(int j = 0; j < m; j++)
        {
            if(i == j && matriz[i][j] == 1)
            {
                lazos++;
            }
            if(matriz[i][j] == 1)
            {
                aristas++;
                grado[i]++;
            }
        }
    }

    cout << "Numero de lazos: " << lazos << endl;
    cout << "Numero de aristas: " << aristas << endl;

    for(int i = 0; i < n; i++)
    {
        cout << "Grado del vertice " << char('A' + i) << ": " << grado[i] << endl;
    }

    return 0;
}

File: modelos ESPE/P3_D1_NRC9901_Reyes_David/python/P3_D1_NRC9901_Reyes_David_No_1.py
import random

n = int(input("Ingrese el tamano de la matriz (nxm): "))
m = int(input())

matriz = [[0 for j in range(m)] for i in range(n)]
for i in range(n):
    for j in range(m):
        matriz[i][j] = random.randint(0, 1)

print("   ", end="")
for i in range(m):
    print(chr(ord('A') + i), end=" ")
print()

for i in range(n):
    print(chr(ord('A') + i), "  ", end="")
    for j in range(m):
        print(matriz[i][j], end=" ")
    print()

lazos = 0
aristas = 0
grado = [0] * n

for i in range(n):
    for j in range(m):
        if i == j and matriz[i][j] == 1:
            lazos += 1
        if matriz[i][j] == 1:
            aristas += 1
            grado[i] += 1

print("Numero de lazos: ", lazos)
print("Numero de aristas: ", aristas)

for i in range(n):
    print("Grado del vertice ", chr(ord('A') + i), ": ", grado[i])

